theano/tensor/signal/pool.py

```diff
@@ -255,6 +255,20 @@ def pool_3d(input, ws=None, ignore_border=None, stride=None, pad=(0, 0, 0),
     return output
 
 
+def spp_pooling(image, out_dimension, image_shape):
+
+    input_size = image_shape[2:]
+    pooled_data_list = []
+    for pool_dim in out_dimension:
+        pool_size = tuple((i + pool_dim - 1) // pool_dim for i in input_size)
+        stride_size = tuple((i // pool_dim) for i in input_size)
+        pooled_part = pool_2d(image, pool_size, ignore_border=True, st=stride_size, padding=(0, 0), mode='max')
+        pooled_part = pooled_part.flatten(3)
+        pooled_data_list.append(pooled_part)
+
+    return tensor.concatenate(pooled_data_list, axis=2)
```

Comment:
Yeah, my bad I wasn't clear in my previous comment. Even the stride is dependent on the pooling_size (which indeed is dependent on the input layer's size). This causes the problem as stride seems to take only fixed values.